### DISCUSSION TITLE
Using LLVM/Clang's MemorySanitizer + AddressSanitizer

### DISCUSSION THREAD

**Discussion by aliireza** [tbarbette/fastclick#307] [Ideas]
Hi,

I think we should start testing FastClick with LLVM/Clang's [MemorySanitizer](http://clang.llvm.org/docs/MemorySanitizer.html) and [AddressSanitizer](http://clang.llvm.org/docs/AddressSanitizer.html) and perhaps integrating them with Travis. 

I opened this issue to remember it :-p 

Alireza

**Comment by tbarbette** [comment]
Yes, good idea! However I would not file this as a bug, but everything it finds will :p

**Comment by pallas** [comment]
You'll want to disable code that pools or shares memory, e.g. every skb clone that might just add a ref should instead return a new skb. It's been a while since I looked at the codebase but IIRC there is stuff like that in string handling too. Otherwise the lifetime of these allocations is long and wild and ASAN / MSAN will take much longer to flag errors.
